src/irc/BridgedClient.ts

```diff
@@ -58,16 +58,34 @@ export interface BridgedClientLogger {
 
 export const illegalCharactersRegex = /[^A-Za-z0-9\]\[\^\\\{\}\-`_\|]/g;
 
+export enum BridgedClientStatus {
+    CREATED,
+    CONNECTING,
+    CONNECTED,
+    DEAD,
+    KILLED,
+}
+
+interface NotConnected {
```

Comment:
I'm not really convinced by these Connected/NotConnected interfaces. I'd rather status,client,inst were all top level. If we must wrap it, does ConnectionInstance suit that purpose as a wrapper for the other two values?

Reply:
this allows the the removal of the ugly `null` checks that were there before on `inst` and `unsafeClient`.

It's a pattern to get rust like enums, that contain values in some variants.

Reply:
I am not sure how you would want to use ConnectionInstance to achieve the same level of type safety (Ensuring that a ConnectionInstance is guaranteed to be there if the status is `CONNECTED`).

Reply:
In the end I evaluated it again and I'm going to go with the approach, but I'm not going to expose client/inst outside of `BridgedClient`. Code that makes use of those will use methods on BridgedClient and we can do some safety checks there.